numba/core/typing/arraydecl.py

```diff
@@ -474,14 +474,14 @@ def resolve_ravel(self, ary, args, kws):
         # Only support no argument version (default order='C')
         assert not kws
         assert not args
-        return signature(ary.copy(ndim=1, layout='C'))
+        return signature(ary.copy(ndim=1, layout='C', readonly=False))
```

Comment:
I'm not sure that this is correct? I think the mutability of the array resulting from `ravel()` would be determined by whether a copy is needed?
```pycon
In [35]: import numpy as np

In [36]: a = np.ones((2, 3))

In [37]: a.flags.writeable = False

In [38]: a.flags
Out[38]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : True
  WRITEABLE : False
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

In [39]: a_c = np.ravel(a)

In [40]: a_c.flags
Out[40]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : False
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

In [41]: a_f = np.ravel(a, "F")

In [42]: a_f.flags
Out[42]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

In [43]: a.ravel('F').flags
Out[43]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

In [44]: a.ravel('C').flags
Out[44]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : False
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

Reply:
Note that in Numba, it appears that no copy is made if the array is C-contiguous, and a copy is made if it is not C-contiguous: https://github.com/numba/numba/blob/main/numba/np/arrayobj.py#L1992

Reply:
Okay, updated the code to be sensitive to this, and added a unit test to verify.